### PR TITLE
Add endpoint for PXE Image Types

### PR DESCRIPTION
**Pull request by lfu**
Fixes https://github.com/ManageIQ/manageiq-api/issues/908

@miq-bot add_label enhancement, kasparov/no

### CODE REVIEW

**Comment by lfu** [comment]
```
GET /api/pxe_image_types

{
    "name": "pxe_image_types",
    "count": 4,
    "subcount": 4,
    "pages": 1,
    "resources": [
        {
            "href": "http://localhost:3000/api/pxe_image_types/10000000000001"
        },
        {
            "href": "http://localhost:3000/api/pxe_image_types/10000000000002"
        },
        {
            "href": "http://localhost:3000/api/pxe_image_types/10000000000003"
        },
        {
            "href": "http://localhost:3000/api/pxe_image_types/10000000000004"
        }
    ],
    "actions": [
        {
            "name": "query",
            "method": "post",
            "href": "http://localhost:3000/api/pxe_image_types"
        },
        {
            "name": "create",
            "method": "post",
            "href": "http://localhost:3000/api/pxe_image_types"
        },
        {
            "name": "edit",
            "method": "post",
            "href": "http://localhost:3000/api/pxe_image_types"
        },
        {
            "name": "delete",
            "method": "post",
            "href": "http://localhost:3000/api/pxe_image_types"
        }
    ],
    "links": {
        "self": "http://localhost:3000/api/pxe_image_types?offset=0",
        "first": "http://localhost:3000/api/pxe_image_types?offset=0",
        "last": "http://localhost:3000/api/pxe_image_types?offset=0"
    }
}
```

```
GET /api/pxe_image_types/10000000000002

{
    "href": "http://localhost:3000/api/pxe_image_types/10000000000002",
    "id": "10000000000002",
    "name": "ESX",
    "provision_type": "host",
    "actions": [
        {
            "name": "edit",
            "method": "post",
            "href": "http://localhost:3000/api/pxe_image_types/10000000000002"
        },
        {
            "name": "delete",
            "method": "post",
            "href": "http://localhost:3000/api/pxe_image_types/10000000000002"
        },
        {
            "name": "delete",
            "method": "delete",
            "href": "http://localhost:3000/api/pxe_image_types/10000000000002"
        }
    ]
}
```

```
POST /api/pxe_image_types
{
    "name": "foo",
    "provision_type": "host"
}

{
    "results": [
        {
            "href": "http://localhost:3000/api/pxe_image_types/10000000000005",
            "id": "10000000000005",
            "name": "foo",
            "provision_type": "host"
        }
    ]
}
```

```
POST /api/pxe_image_types/10000000000006
{
    "action": "delete"
}

{
    "success": true,
    "message": "pxe_image_types id: 10000000000006 deleting",
    "href": "http://localhost:3000/api/pxe_image_types/10000000000006"
}
```

```
DELETE /api/pxe_image_types/10000000000005
```

File: spec/requests/pxe_image_types_spec.rb [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+  end
+
+  describe 'DELETE /api/pxe_image_types/:id' do
+    it "deletes a pxe image tpye with an appropriate role" do
+      api_basic_authorize(action_identifier(:pxe_image_types, :delete))
+      delete(api_pxe_image_type_url(nil, image_type))
+      expect(response).to have_http_status(:no_content)
+    end
+
+    it 'forbids deleting a pxe image tpye without an appropriate role' do
```
Comment (abellotti) [inline spec/requests/pxe_image_types_spec.rb:136]: tpye -> type.   here and others, need a global replace.

**Review by abellotti — APPROVED** [review]

**Comment by abellotti** [comment]
LGTM!! Thanks @lfu for the API Enhancement. 👍

File: config/api.yml [hunk 2]
```diff
@@ -2869,6 +2869,38 @@
         :identifier: miq_request_control
       - :name: edit
         :identifier: miq_request_edit
+  :pxe_image_types:
+    :description: PXE Image Types
+    :identifier: pxe_image_type_accord
+    :options:
+    - :collection
+    :verbs: *gpd
```
Comment (Fryguy) [inline config/api.yml:2877]: Should this be `gpppd`?  @abellotti ?

**Review by Fryguy — APPROVED** [review]
LGTM - one small question re: scope of the verbs

File: config/api.yml (hunk 2 above)
Comment (abellotti) [reply config/api.yml:2877]: It can be, essentially adding the PATCH and PUT, if we do, then the specs should be extended to include those, that's all.

Comment (lfu) [reply config/api.yml:2877]: What is the criteria for adding `PATCH` and `PUT`? Just curious.

Comment (Fryguy) [reply config/api.yml:2877]: I generally think we should do post/patch/put as a group.  I think the better question is "What is the criteria for NOT adding `PATCH` and `PUT`"?

**Review by gtanzillo — APPROVED** [review]

**Pull request merged**